images/base/Dockerfile

```diff
@@ -55,6 +55,14 @@ ARG CNI_PLUGINS_AMD64_SHA256SUM="962100bbc4baeaaa5748cdbfce941f756b1531c2eadb290
 ARG CNI_PLUGINS_ARM64_SHA256SUM="ef17764ffd6cdcb16d76401bac1db6acc050c9b088f1be5efa0e094ea3b01df0"
 ARG CNI_PLUGINS_PPC64LE_SHA256SUM="5bd3c82ef248e5c6cc388f25545aa5a7d318778e5f9bc0a31475361bb27acefe"
 
+# Configure containerd-fuse-overlayfs snapshotter binary from upstream
+ARG CONTAINERD_FUSE_OVERLAYFS_VERSION="1.0.2"
+ARG CONTAINERD_FUSE_OVERLAYFS_TARBALL="v${CONTAINERD_FUSE_OVERLAYFS_VERSION}/containerd-fuse-overlayfs-${CONTAINERD_FUSE_OVERLAYFS_VERSION}-linux-${TARGETARCH}.tar.gz"
+ARG CONTAINERD_FUSE_OVERLAYFS_URL="https://github.com/containerd/fuse-overlayfs-snapshotter/releases/download/${CONTAINERD_FUSE_OVERLAYFS_TARBALL}"
```

Comment:
Will this eventually be merged in? It seems like this is crucial enough to bake in 🤔

Reply:
No, the containerd/containerd repo does not accept a new built-in snapshotter.

Reply:
> No, the containerd/containerd repo does not accept a new built-in snapshotter.

That seems unfortunate, especially with it being essentially required for rootless. I hope they may reconsider for this case.

> If having a separate binary is not comfortable, maybe we can consider patching kind-ci/containerd-nightlies. (Just adding import _ "github.com/containerd/fuse-overlayfs-snapshotter/plugin" line)

We probably don't want to patch containerd sources, but maybe worth considering if we can do it with our own stub source importing or something.

Users building images for compliance (e.g. EKS AIUI) will still need to update their builds to fix this though. It's less that we need to add a binary to the image and more that there's another unverified binary they may need to notice and start overriding + building. If we keep doing that it makes updating kind a riskier proposition.

It's also a little annoying to need to run more services though / extra entrypoint complexity though and I imagine the binary size is smaller when merged in. But it is *not* a deal breaker. It's still more important to make sure kind works.